msg.go

```diff
@@ -603,71 +605,91 @@ func packRR(rr RR, msg []byte, off int, compression compressionMap, compress boo
 
 // UnpackRR unpacks msg[off:] into an RR.
 func UnpackRR(msg []byte, off int) (rr RR, off1 int, err error) {
-	h, off, msg, err := unpackHeader(msg, off)
-	if err != nil {
-		return nil, len(msg), err
+	if off < 0 || off > len(msg) {
+		return nil, off, &Error{err: "bad off"}
 	}
 
-	return UnpackRRWithHeader(h, msg, off)
+	s := cryptobyte.String(msg[off:])
+	rr, err = unpackRR(&s, msg)
+	if err != nil {
+		// Keep existing behaviour of returning len(msg) here on error.
+		return rr, len(msg), err
```

Comment:
@miekg Do we want to keep this behaviour? It's not actually documented here unlike on `UnpackDomainName`.

Reply:
make it return `off` here too? Seems more useful than `len(msg)` in any case.